components/_twig-components/functions/blockname.function.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * @file
+ * Add "classes" function for Pattern Lab.
+ */
+
+$function = new Twig_SimpleFunction('classes', function ($base_class, $modifiers = array(), $blockname = '') {
+  $classes = array();
+  $classes[] = $base_class;
+  if (isset($modifiers) && is_array($modifiers)) {
+    foreach ($modifiers as $modifier) {
+      $classes[] = $base_class . '--' . $modifier;
+    };
+  }
+  if ($blockname) {
+    $classes[] = $blockname . '__' . $base_class;
+  }
```

Comment:
in the `blockname` conditional, I wonder if the modifiers loop should be repeated. It's a pattern I rely on frequently enough to standardize. Then you can target a modified component within a parent like so:

```scss
// abstract
.blockname__baseclass--modifier {}

// example
.bios__profile--badge-webchef {}
```